ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java

```diff
@@ -242,6 +242,16 @@ ReceiveResponse getReceiveResponse(final String key) {
     return receiveResponse;
   }
 
+  boolean skipContractExecution(final MessageFrame messageFrame) {
+    return isReadOnly(messageFrame) || isMining(messageFrame);
+  }
+
+  boolean isReadOnly(final MessageFrame messageFrame) {
+    // If there's no PrivateMetadataUpdater, the precompile has likely been called as part of a
```

Comment:
I don't like the "is likely" :-) We should know!

Is that correct? :  "If there's no PrivateMetadataUpdater, that means that the precompile has not been called through the PrivacyBlockProcessor. That means that the precompile has been called because a PMT is being simulated. The execution of the precompile has no influence on the outcome of that simulation."